module/Solr/src/Solr/Listener/JobEventSubscriber.php

```diff
@@ -49,154 +73,101 @@ public function __construct(Manager $manager)
     public function getSubscribedEvents()
     {
         return [
+            Events::preUpdate,
             Events::postUpdate,
-            Events::postPersist,
+            Events::postFlush
         ];
     }
-    public function consoleIndex(Job $job)
-    {
-        $this->updateIndex($job);
-    }
+    
     /**
-     * Handle doctrine post persist event
-     *
      * @param LifecycleEventArgs $eventArgs
+     * @since 0.27
      */
-    public function postPersist(LifecycleEventArgs $eventArgs)
+    public function preUpdate(PreUpdateEventArgs $eventArgs)
     {
         $document = $eventArgs->getDocument();
-        $this->updateIndex($document);
+        
+        // check for a job instance
+        if (!$document instanceof Job) {
+            return;
+        }
+        
+        // check if the status has been changed
+        if (!$eventArgs->hasChangedField('status')) {
+            return;
+        }
+            
+        // check if the job is active
+        if ($document->isActive()) {
+            // mark it for commit
+            $this->add[] = $document;
+        } else {
+            // mark it for delete
+            $this->delete[] = $document;
+        }
```

Comment:
No, the job should **only** be removed from solr index, when the state changed to "inactive".

Reason: 
Job could be edited after publishing, and these edition must be reviewed, but the job WITHOUT modification should still be available in the jobboard.